### PR TITLE
move extracting requester to utils

### PR DESCRIPTION
**Pull request by kswiatek92**
service account can be None
but user will always be set (user or anonymous)
previous `info.context.user or info.context.service_account` always returned user

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/graphql/product/resolvers.py [hunk 1]
```diff
@@ -141,9 +142,7 @@ def resolve_products(
     **_kwargs,
 ):
 
-    user = info.context.user
```
Comment (fowczarek) [inline saleor/graphql/product/resolvers.py]: Probably we should use `get_requester_from_context` in almost all places where `user = info.context.user` appear in code. All query and mutation should work for user and service account with proper permissions.

**Comment by django-queries** [comment]
Here is the report for 07c977be21bc735e045b817581e0c903bdc312be (kswiatek92/saleor @ move-getting-requester-to-utils)
Base comparison is a0c379ef7a81073c509d11907b2bef341eb12da2.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	         50	         50	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         14	         14	              2
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         16	         16	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/product/resolvers.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/product/resolvers.py]: Sometimes we want to get only user and sometimes user or service account. This is the second case. For getting user from the context we could also have a helper.

File: saleor/graphql/utils.py [hunk 2]
```diff
@@ -183,3 +183,9 @@ def create_jwt_payload(user, context=None):
     payload["is_staff"] = user.is_staff
     payload["is_superuser"] = user.is_superuser
     return payload
+
+
+def get_requester_from_context(context):
```
Comment (maarcingebala) [inline saleor/graphql/utils.py]: Maybe `get_user_or_service_account_from_context`? A bit long but I assume we could also have `get_user_from_context` and just `get_service_account_from_context`.

**Pull request merged**